Automattic/woocommerce-payments
# Activate fraud services before WPCOM/Jetpack connection and account onboarding

Fixes #7517 

#### Changes proposed in this Pull Request

Title: Activate fraud tracking before onboarding and/or WPCOM/Jetpack connection

Description:

Until now, we have provided the fraud services configuration only through the account data (the `fraud_services` entry). This meant that we could not activate them prior to onboarding. This hampers our accounts fraud detection capabilities.

We now take advantage of a new server API endpoint (`accounts/fraud_services`) that will provide the public, account-agnostic fraud services configuration so we can properly initialize them without the need for an account or a Jetpack connection (the endpoint is public). This refers right now to Sift activation.

The fraud services configuration in the account data is still used and _takes precedence_ over the one from the public endpoint.

We will activate the Sift JS SDK tracking in all WooCommerce admin pages and the WooPayments Connect page. 

#### Testing instructions

* Checkout the PR branch on your local client installation.
* Checkout the pair server PR branch (https://github.com/Automattic/woocommerce-payments-server/pull/4045 ) on your local server installation.
* Delete any active account in your local server so the Payments Connect page is available.
* Use [the Jetpack Debug Tools plugin](https://github.com/Automattic/jetpack-debug-helper) to disconnect your client from WPCOM:
  * Activate the `Broken Token Utilities`
  * Go to **Jetpack Debug > Broken Token** 
  * Click on **Store these options** to make a backup of your current options so you can restore them later on
  * Click on **Clear blog token**, **Clear user tokens**, and **Clear the Blog ID** so the Jetpack connection is as broken as it gets
  * Go to WCPay Dev Tools and confirm this by seeing "The store can't talk with the WCPay server!" in the Store Details sidebar 
* Open your browser Dev Tools and go to the Networking tab
* Go to the **Payments Connect** page and confirm that you have a request for `https://cdn.sift.com/s.js` (search by `sift`). You should also have a request to `https://hexagon-analytics.com`
* Go to **WooCommerce > Home** and confirm that you see the same two requests in your browser Networking view
* Go to **Jetpack Debug > Broken** token and click on **Restore from stored options**
* Make sure you activate Progressive Onboarding through the WCPay Dev Tools
* Start the Stripe event listener on your server by running `npm run listen`
* Click on **Finish setup** on the Payments Connect page to go to the PO wizard.
* Confirm that you see the same two requests in your browser Networking view
* Finish the onboarding
* Go to **Payments Overview** and confirm that you see the same two requests in your browser Networking view
* In your browser console, check `wcpaySettings.fraudServices.sift`. It should have entries like these:
```
beacon_key: "XXXXX"
session_id: "st_XXXXXXXXX_{WP USERID}"
user_id: "acct_XXXXXX"
```
* Make sure the `user_id` in `wcpaySettings.fraudServices.sift` matches the one in WCPay Dev Tools' Store Details sidebar
* Go to **WooCommerce > Home** and confirm that you see the same two requests in your browser Networking view
* Add a product to cart, go to the **Checkout** page, and confirm that you see the same two requests in your browser Networking view
* In your browser console, check `wcpayConfig.fraudServices.sift`. It should have entries like these:
```
beacon_key: "XXXXX"
session_id: "st_XXXXXXXXX_{WP USERID}"
user_id: "cus_XXXXXX"
```
* In an incognito browser session, add a product to cart, go to the **Checkout** page, and confirm that you see the same two requests in your browser Networking view

-------------------


**Post merge**
